core/src/test/scala/unit/kafka/server/ReplicaFetcherThreadTest.scala

```diff
@@ -812,19 +813,27 @@ class ReplicaFetcherThreadTest {
       time = new SystemTime(),
       quota = null,
       leaderEndpointBlockingSend = Some(mockBlockingSend))
+    thread.start()
 
     val previousLevel = LogCaptureAppender.setClassLoggerLevel(thread.getClass, Level.DEBUG)
     val logCaptureAppender = LogCaptureAppender.createAndRegister()
 
     try {
       thread.initiateShutdown()
 
-      val event = logCaptureAppender.getMessages.find(e => e.getLevel == Level.DEBUG
-        && e.getRenderedMessage.contains(s"Fail to close leader endpoint $mockBlockingSend after initiating replica fetcher thread shutdown")
+      val event = logCaptureAppender.getMessages.find(e => e.getLevel == Level.ERROR
```

Comment:
This is kind of icky. I wonder if we should consider adding a field to `ShutdownableThread` to save the failure exception?

Reply:
Agree this is not nice. The first log entry comparison was added under KAFKA-7464. And I extended that to add another of the same type. I think we want the test to ensure that exceptions are not propagated to the caller and that we are closing the blockingSend instance. And that is tested by this subset of that test:
```
thread.initiateShutdown()
thread.awaitShutdown()
verify(mockBlockingSend)
```
We dont really need to verify the log entry at all for this. I suppose the log entry comparison checks that the test is correct by verifying that it did inject that exception, but not sure we really need to do that. It feels unnecessary to store away the exception in `ShutdownableThread` just to check this. What do you think?

Reply:
Yes, that makes sense to me.

Reply:
@hachikuji Thank you, updated.